opencodeblocks/scene/ipynb_conversion.py

```diff
@@ -0,0 +1,56 @@
+""" Module for converting ipynb data to ipyg data """
+
+from typing import OrderedDict, List
+
+import json
+
+MARGIN: float = 50
+
+def ipynb_to_ipyg(data: OrderedDict) -> OrderedDict:
+    """ Convert ipynb data (ipynb file, as ordered dict) into ipyg data (ipyg, as ordered dict) """
+
+    dataid: int = 0 # TODO : give a proper id
+
+    blocks: List[OrderedDict] = get_blocks(data)
+
+    return {
+        "id": dataid,
+        "blocks": blocks,
+        "edges": []
+    }
+
+def get_blocks(data: OrderedDict) -> List[OrderedDict]:
+    """ Get the blocks corresponding to a ipynb file, returns them in the ipyg ordered dict format """
+
+    if "cells" not in data:
+        return []
+    
+    blocks: List[OrderedDict] = []
+
+    for cell in data["cells"]:
+        if "cell_type" not in cell or cell["cell_type"] != "code":
+            pass # TODO : support markdown
+        else:
+            # Load the default empty block
+            # TODO : add something in case the user renames / removes the empty block / changes it too much ?
+            data: OrderedDict = {}
+            with open("blocks/empty.ocbb", 'r', encoding='utf-8') as file:
```

Comment:
The empty block might change places in the future. The data fields should be set from scratch.
```python
data = OrderedDict([
  ("id",0),
  # ...
])
```
You can checkout the `serialize` in `block.py` as an example

Reply:
27022fbe2f4960fcc0c23b90b8ec147a8c2c060d
You can do something similar to what you did in this commit.

Reply:
In the future, the serialization process will be more robust and we'll be able to not set some fields like splitter and enjoy sane defaults.